packages/prompts/src/index.ts

```diff
@@ -87,6 +88,38 @@ export const text = (opts: TextOptions) => {
 	}).prompt() as Promise<string | symbol>;
 };
 
+export interface PasswordOptions {
+	message: string;
+	mask?: string;
+	validate?: (value: string) => string | void;
+}
+export const password = (opts: PasswordOptions) => {
+	return new PasswordPrompt({
+		validate: opts.validate,
+		mask: opts.mask,
+		render() {
+			const title = `${color.gray(S_BAR)}\n${symbol(this.state)}  ${opts.message}\n`;
+			const value = this.valueWithCursor;
+			const masked = this.masked;
+
+			switch (this.state) {
+				case 'error':
+					return `${title.trim()}\n${color.yellow(S_BAR)}  ${masked}\n${color.yellow(
+						S_BAR_END
+					)}  ${color.yellow(this.error)}\n`;
+				case 'submit':
+					return `${title}${color.gray(S_BAR)}  ${color.dim(masked)}`;
+				case 'cancel':
+					return `${title}${color.gray(S_BAR)}  ${color.strikethrough(color.dim(masked ?? ''))}${
+						masked?.trim() ? '\n' + color.gray(S_BAR) : ''
```

Comment:
Haven't tested, but is `trim()` legit in this case? A space would be a valid password character, no?

Reply:
Or rather, would this ever trim anything as spaces would be replaced by the masking character?

Reply:
I followed the implementation of the `text` prompt (https://github.com/natemoo-re/clack/blob/main/packages/prompts/src/index.ts#L74) here, but I guess a space at the end is usually trimmed when you provide a password? Anyway, it won't change the underlying value, as this part of the code only prints stuff to the user's console. But let me know if I should remove the `trim()`.

Reply:
> but I guess a space at the end is usually trimmed when you provide a password

I think not. I'd say that's bad practice.

> Anyway, it won't change the underlying value, as this part of the code only prints stuff to the user's console.

Yes, but what I'm saying is will this ever actually change anything? If not, I'd argue we remove it. If spaces are replaced with the mask character there will never be anything to trim.

Reply:
Agreed, the trim can be removed in this case since passwords should be passed literally